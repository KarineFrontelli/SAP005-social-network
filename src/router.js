// Este é seu ponto de entrada da sua aplicação
import { Home } from './pages/home/index.js';
import { Login } from './pages/login/index.js';
// import { onNavigate } from './utils/history.js';

const routes = {
  '/home': Home,
  '/': Login,
};

const container = document.querySelector('#root');
const validateHash = (hash) => (hash === '' ? '/' : hash.replace('#', ''));

export const renderPage = () => {
  const page = validateHash(window.location.pathname);
  container.innerHTML = '';
  container.appendChild(routes[page]());
};

const init = () => window.addEventListener('hashchange', renderPage);

window.addEventListener('load', () => {
  renderPage();
  init();
});

window.addEventListener('popstate', renderPage);
