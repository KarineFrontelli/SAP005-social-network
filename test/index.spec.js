import * as services from '../src/services/index.js';
import { Login } from '../src/pages/login/index.js';
import { renderPage } from '../src/router.js';
// import { Home } from '../src/pages/home/index.js';

services.likeFirebase = jest.fn(() => Promise.resolve(true));
services.editPost = jest.fn(() => Promise.resolve(true));
services.excluirPostBanco = jest.fn(() => Promise.resolve(true));
services.Google = jest.fn(() => Promise.resolve(true));
services.Facebook = jest.fn(() => Promise.resolve(true));

describe('Render página de login', () => {
  it('carrega a página', () => {
    expect(Login()).toMatchSnapshot();
  });

  it('click no botão de login com o Google', () => {
    const pagLogin = Login();
    pagLogin.querySelector('#btnGoogle').dispatchEvent(new Event('click'));
    expect(services.Google).toHaveBeenCalled();
  });

  it('click no botão de login com o Facebook', () => {
    const pagLogin = Login();
    pagLogin.querySelector('#btnFacebook').dispatchEvent(new Event('click'));
    expect(services.Facebook).toHaveBeenCalled();
  });
});

describe('Testar renderPage', () => {
  it('é uma função', () => {
    expect(typeof renderPage).toBe('function');
  });
});

describe('Testar login', () => {
  it('é uma função', () => {
    expect(typeof Login).toBe('function');
  });
});

describe('Testar like', () => {
  it('é uma função', () => {
    expect(typeof services.likeFirebase).toBe('function');
  });
});

describe('Testar edit', () => {
  it('é uma função', () => {
    expect(typeof services.editPost).toBe('function');
  });
});

describe('Testar excluir', () => {
  it('é uma função', () => {
    expect(typeof services.excluirPostBanco).toBe('function');
  });
});
