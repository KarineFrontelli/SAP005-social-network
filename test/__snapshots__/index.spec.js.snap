// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render página de login carrega a página 1`] = `
<div>
  
    
  <section
    class="login"
  >
    
      
    <div
      class="logo"
    >
      
        
      <img
        alt="logo da  página"
        class="img-logo"
        src="../img/Logo.jpeg"
      />
      
      
    </div>
    
      
    <form
      class="order-login"
    >
      
        
      <div>
        
          
        <input
          class="input-login"
          id="inputEmail"
          placeholder="Email"
          type="email"
        />
        
        
      </div>
      
        
      <div>
        
          
        <input
          class="input-login"
          id="inputSenha"
          placeholder="Senha"
          type="password"
        />
        
        
      </div>
      
          
      <button
        class="btn-Login"
        id="btnEmail"
      >
        Login
      </button>
      
          
      <p
        class="login-com"
      >
         Ou faça login com...
      </p>
      
      
    </form>
    
      
    <div
      class="outros-logins"
    >
      
        
      <button
        class="btn-Google"
        id="btnGoogle"
      >
        Google
      </button>
      
        
      <button
        class="btn-Facebook"
        id="btnFacebook"
      >
        Facebook
      </button>
      
      
    </div>
    
      
    <div
      <button=""
      class="criar-conta"
      id="criarCadastro"
    >
      Não é membro? Criar conta
      
    </div>
    
    
  </section>
  
  
</div>
`;
